internal/cluster/cluster.go

```diff
@@ -211,6 +216,64 @@ func (m *Manager) DeregisterCluster(ctx context.Context, c *common.Cluster) erro
 	return err
 }
 
+func (m *Manager) reportValidationFailedMetrics(ctx context.Context, c *common.Cluster, ocpVersion, emailDomain string) error {
+	log := logutil.FromContext(ctx, m.log)
+	var validationRes validationsStatus
+	if c.ValidationsInfo != "" {
+		if err := json.Unmarshal([]byte(c.ValidationsInfo), &validationRes); err != nil {
+			log.WithError(err).Errorf("Failed to unmarshal validations info from cluster %s", *c.ID)
+			return err
+		}
+	}
+	for _, vRes := range validationRes {
+		for _, v := range vRes {
+			if v.Status == ValidationFailure {
+				m.metricAPI.ClusterValidationFailed(ocpVersion, emailDomain, models.ClusterValidationID(v.ID))
+			}
+		}
+	}
+	return nil
+}
+
+func (m *Manager) reportValidationStatusChanged(ctx context.Context, c *common.Cluster,
+	newValidationRes, currentValidationRes validationsStatus) {
+	for vCategory, vRes := range currentValidationRes {
```

Comment:
Validations might change, I suggest to iterate over the newValidationRes

Reply:
That what I was doing at first in host  validations, but then it broke old clusters that didn't contains some validations added later on, so after discussing it, I decide to iterate on the old validations, then the RefreshStatus will update them in DB and they will be iterated on in the next call to RefreshStatus

Reply:
So now it will break if we remove a validation?
Consider handling the case that a validation exists in one ValidationRes and not the other.

Reply:
> So now it will break if we remove a validation?

Yes, I don't see how we can do it otherwise.

Reply:
DB may contain clusters from older deployment.
The code should handle it.
Why not check if the key exists in the map before comparing the value?

Reply:
> DB may contain clusters from older deployment. The code should handle it.

This is exactly the reason why I iterate over existing validations instead of new validations that may not exist in old clusters.